### DISCUSSION TITLE
PAM setup in the new Container/Podman Install

### DISCUSSION THREAD

**Discussion by RenownTwiz** [uyuni-project/uyuni#8675] [Q&A] [answered]
I started reviewing the new container version.  Got everything installed on openSUSE MicroOS VERSION_ID="20240428", GUI is up and working.  Was looking at setting up user accounts and on the current VM install we use the "Use PAM" checkbox.  The Documentation all still points to what I believe is the original VM install of PAM adding pam.d file and editing rhn.conf.  It seems to me that in this new evolution this would have to be done inside the container not at the MicroOS level?  I don't even see SSSD installed in MicroOS be default.  Has anyone tried to get this working on the Container base install yet?  Any initial thoughts on if this feature is even supported yet in the container based approach?

vi /etc/pam.d/susemanager

#%PAM-1.0
auth     include        common-auth
account  include        common-account
password include        common-password
session  include        common-session

pam-config -a --sss

vi /etc/rhn/rhn.conf
pam_auth_service = susemanager

**Comment by rjmateus** [comment, marked as answer]
Hey,
Pam authentication support is already developed. I'm not sure if is already part of the latest image.
Implementaion PR:
- https://github.com/uyuni-project/uyuni/pull/8537
- https://github.com/uyuni-project/uyuni/pull/8632/files

Documentation PR:
- https://github.com/uyuni-project/uyuni-docs/pull/2935

Basically you need to enter the container terminal (`mgrctl term`) and then configure the configuration for sssd.

**Reply by RenownTwiz** [reply]
Thanks for the link and the answer.  I'm not sure I've figured out how to tell what is or is not included in registry.opensuse.org/uyuni/server:latest but it doesn't seem to have any sssd stuff baked into the container just yet.  I'm just guessing here, but maybe "latest" is built from branch Uyuni-2024.03 right now?  It looks like the commits in your links are not yet in that branch.  My guess is if I just wait for a bit and retest it will be baked into the image as you suggest.  Thanks very much for answering my question.

**Reply by mbussolotto** [reply]
Hi @RenownTwiz,
The stable version (the one avaiable from registry.opensuse.org/uyuni/server:latest) does not contains the PAM setup changes, they willl be present from next stable (they're already merged in master as described ^^^)

**Reply by mbussolotto** [reply]
besides the one mentioned, this is the last PR about PAM we merged a week ago https://github.com/uyuni-project/uyuni/pull/8648/
